contracts/core/modules/IssuanceOrderModule.sol

```diff
@@ -405,48 +327,194 @@ contract IssuanceOrderModule is
+     *
+     * maker                            Issuance order maker
+     * taker                            Issuance order taker
+     * makerToken                       Address of maker token used in exchange orders
+     * makerAssetAmount                 Amount of issuance order maker token to use on this exchange
+     * orderCount                       Expected number of orders to execute
+     * fillQuantity                     Quantity of Set to be filled
+     * attemptedfillQuantity            Quantity of Set taker attempted to fill
+     *
+     * @param  _addresses               [maker, taker, makerToken]
+     * @param  _values                  [makerAssetAmount, orderCount, fillQuantity, attemptedFillQuantity]
+     * @param  _exchange                Address of exchange wrapper being called
+     * @param  _bodyData                Arbitrary bytes data for orders to be executed on exchange
+     */
+    function callExchange(
+        address[3] _addresses,
+        uint256[4] _values,
+        address _exchange,
+        bytes _bodyData
+    )
+        private
+    {
+        // Call Exchange
+        address[] memory componentFillTokens = new address[](_values[1]);
+        uint256[] memory componentFillAmounts = new uint256[](_values[1]);
+        (componentFillTokens, componentFillAmounts) = IExchangeWrapper(_exchange).exchange(
+            _addresses,
+            _values,
+            _bodyData
         );
 
-        uint256[] memory requiredBalances = new uint256[](_order.requiredComponents.length);
+        // Transfer component tokens from wrapper to vault
+        ICore(core).batchDepositModule(
+            _exchange,
+            _addresses[0],
+            componentFillTokens,
+            componentFillAmounts
+        );        
+    }
 
-        // Calculate amount of maker token required
-        uint256 requiredMakerTokenAmount = OrderLibrary.getPartialAmount(
-            _order.makerTokenAmount,
-            _fillQuantity,
-            _order.quantity
+    /**
+     * Check exchange orders acquire correct amount of tokens and taker doesn't over use
+     * the issuance order maker's tokens
+     *
+     * @param  _order                       IssuanceOrder object containing order params
+     * @param  _requiredBalances            Array of required balances for each component
+                                            after exchange orders are executed
+     * @param  _requiredMakerTokenAmount    Max amount of maker token used to source tokens
+     * @param  _makerTokenAmountUsed        Amount of maker token used to source tokens
+     */
+    function checkRequiredTokenBalances(
+        OrderLibrary.IssuanceOrder _order,
+        uint256[] _requiredBalances,
+        uint256 _requiredMakerTokenAmount,
+        uint256 _makerTokenAmountUsed
+    )
+        private
+        view
+    {
+        // Verify maker token used is less than amount allocated that user signed
+        require(
```

Comment:
https://blog.codinghorror.com/curlys-law-do-one-thing/

The name of this function is `checkRequiedTokenBalances` - thus I would expect it to just check the balances of the required tokens. I'd move this require up on the level of the `settleOrder` function.